pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -96,16 +121,57 @@ func NewPersistentVolumeRecycler(kubeClient client.Interface, syncPeriod time.Du
 	return recycler, nil
 }
 
+func (recycler *PersistentVolumeRecycler) shouldRecycle(pv *api.PersistentVolume) error {
+	if pv.Spec.ClaimRef == nil {
+		return fmt.Errorf("Volume does not have a reference to claim")
+	}
+	if pv.Status.Phase == api.VolumeReleased {
+		// This is the first attempt to recycle, allow.
+		return nil
```

Comment:
I think this is confusing now.  I understand why you've done this, because the recycler will mark the PV failed if recycling fails and you check for that status below.   But that now means 2 different states represent a need to recycle:  status=Released || status=Failed && !exceedFailedCount.  

`kubectl get pv` would show PVs that failed, but only some are truly Failed.  

We should use Released with the count you're keeping.  When the count is exceeded, the PV is truly failed.